test/pthread/test_futex_wake_all.cpp

```diff
@@ -57,7 +57,7 @@ void *WaitingThread(void *arg)
 
 int main()
 {
-	for(int i = 0; i < NUM_THREADS; ++i)
+	for(intptr_t i = 0; i < NUM_THREADS; ++i)
```

Comment:
Surely the number of threads fits in an `int`? :smile: 

lgtm regardless, but I'm curious if this fixed a specific issue? Maybe that the higher bits were undefined?

Reply:
The specific issue here and elsewhere is that `i` is cast to a `void*` and passed to pthread_create as the argument to the thread start function.

Casting an `int` to `void*` is find on wasm32 but generates a warning on wasm64.

Reply:
I see, thanks.